src/libxively/control_topic/xi_control_topic_name.c

```diff
@@ -76,17 +77,13 @@ xi_state_t xi_control_topic_create_topic_name( char** subscribe_topic_name,
     xi_state_t local_state = XI_STATE_OK;
 
     local_state = xi_control_topic_allocate_and_snprintf(
-        subscribe_topic_name,
-        xi_globals.str_device_unique_id == NULL ? device_id_is_missing
-                                                : xi_globals.str_device_unique_id,
```

Comment:
I think we should remove it completely. It isn't used at all. Just the setter/getter uses it so it is totally pointless. We should have deleted it already since it just confuses users and enables them to pass two different deviceid/userid which is wrong.
I can do the removal here if you accept this gateway feature will affect the original API as well. Do you agree to remove this parameter from xi_initialize?

Reply:
Sure!  I was actually edging on the case of ignorance, not wanting to keep it around.   
anyway, a new gateway seems to be a good time for a major version number change anyway. :)

Reply:
I did this. Too many files are affected. I would like to do this in separate PR on this same branch. This change would make really hard to check this PR. In one hand xi_initialize function change affects too many files and in some files even the code formatter makes it hard to check changes.

Reply:
:+1: